test/TensorOneDTest.cpp

```diff
@@ -0,0 +1,123 @@
+/*
+ * Copyright (c) 2015 - 2022, Intel Corporation
+ * SPDX-License-Identifier: BSD-3-Clause
+ */
+
+#include "gtest/gtest.h"
+#include "gmock/gmock.h"
+#include "geopm/Exception.hpp"
+
+#include "TensorOneD.hpp"
+
+using geopm::TensorOneD;
+
+class TensorOneDTest : public ::testing::Test
+{
+    protected:
+        void SetUp();
+
+        TensorOneD one, two, three;
+};
+
+void TensorOneDTest::SetUp()
+{
+    one.set_dim(2);
+    two.set_dim(2);
+    three.set_dim(3);
+
+    one[0] = 1;
+    one[1] = 2;
+    two[0] = 3;
+    two[1] = 4;
+    three[0] = 0;
+    three[1] = 1;
+    three[2] = 1;
+}
+
+TEST_F(TensorOneDTest, test_sum)
+{
```

Comment:
Can we have a test when you try to add or subtract two instances of the same object?

```cpp
TensorOneD another_two = one + one;
```

Reply:
Added!